### PR TITLE
chore(vrl): VRL separation part 3 (vector-common)

### PR DESCRIPTION
**Pull request by fuchsnj**
ref: https://github.com/vectordotdev/vector/issues/15501
ref: https://github.com/vectordotdev/vector/issues/3784

This is the 3rd of a series of changes to move VRL to it's own project.

In this part, the `vector-common` dependency was removed from VRL crates. Some of the dependencies were only used in VRL, so they were moved into `vrl-core`. There were some that were used in both. These were also moved into `vrl-core`, and then re-exported by `vector-common`.

### CODE REVIEW

File: lib/vector-core/Cargo.toml [hunk 1]
```diff
@@ -61,6 +61,7 @@ typetag = { version = "0.2.5", default-features = false }
 twox-hash = { version = "1.6.3", default-features = false }
 url = { version = "2", default-features = false }
 value = { path = "../vrl/value", default-features = false, features = ["lua", "toml", "json", "api"] }
+
```
Comment (bruceg) [inline lib/vector-core/Cargo.toml:64]: Oops?

**Review by spencergilbert — APPROVED** [review]
Changes look fine to me, based on my understanding - though I'm not super familiar with any intricacies in this area of the codebase.

**Review by neuronull — APPROVED** [review]
LGTM

**Review by bruceg — APPROVED** [review]

**Comment by fuchsnj** [comment]
> Does this need a re-review, or are you just wrestling with CI?

Just trying to convince CI that it is in fact happy.

**Pull request merged**